### PR TITLE
androidenv: use emulator from the path that makes it find qemu

### PR DESCRIPTION
**Pull request by Ekleog**
###### Description of changes

The emulator looks for qemu in a path relative to itself. So the version in /tools/ is broken, and only the other version in /emulator/ is actually working.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable: manually and only `emulator`, could not find related nixos tests (which makes sense given emulator was crashing failing to find qemu anyway)
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by primeos** [comment]
Not sure why I was requested as reviewer. I did some Android stuff in the past but nowadays I'm only maintaining `android-tools` (and even that is slowing down, unfortunately). I'd recommend pining more active Nixpkgs contributors to the android ecosystem (s. git-log, `git shortlog -sne $file`, etc.). You can also look for the most active contributors to that file here: https://github.com/NixOS/nixpkgs/blob/d0e691c9fee72b55fa4ecfe88e72cdd696e08100/pkgs/development/mobile/androidenv/compose-android-packages.nix ;)

**Comment by Ekleog** [comment]
Well, you looked like you had done non-trivial recent-ish changes to androidenv in the git log, hence my pick :sweat_smile: (the webpage you pointed to has the drawback of also showing people who contributed a very long time ago). Anyway, thank you for the work you have put into androidenv!

**Comment by primeos** [comment]
cc @Artturin @Atemu @vtuan10 (maybe one of you could properly review/test this? (looking at `git log pkgs/development/mobile/androidenv/`))

> Well, you looked like you had done non-trivial recent-ish changes to androidenv in the git log

Looks like I do indeed appear in the log :)
Unfortunately it's around 5 years ago and I don't even use `androidenv` anymore (only need `adb` nowadays and even that only rarely :D).

**Comment by Artturin** [comment]
> The emulator looks for qemu

qemu isn't mentioned in the expression anywhere so i assume its an impurity?
could either of the emulator/emulator be patched to directly use it

File: pkgs/development/mobile/androidenv/compose-android-packages.nix [hunk 1]
```diff
@@ -312,6 +312,11 @@ rec {
           ln -s $i $out/bin
       done
 
+      ${lib.optionalString includeEmulator ''
+        rm $out/bin/emulator
+        ln -s $out/libexec/android-sdk/emulator/emulator $out/bin
+      ''}
```
Comment (Artturin) [inline pkgs/development/mobile/androidenv/compose-android-packages.nix:318]: needs comment

**Comment by Ekleog** [comment]
@Artturin Done :) The emulator uses qemu from its own executable directory with this patch, so AFAIU there's no impurity after that. (I seem to understand that it's using a patched qemu and can't use the system qemu; not sure but it's already building the qemu today anyway)

@primeos Ooh sorry about that! I hadn't thought of checking the date :sweat_smile:

**Comment by Artturin** [comment]
Squash the commits

**Comment by Ekleog** [comment]
Seeing this seems to be the last comment, I'll assume the rest of the PR seems OK with you, and hit the "squash and merge" button. Thank you for your review!

**Pull request merged**